<?xml version="1.0" encoding="utf-8"?>
<launch>
  <node pkg="rosbag" type="play" name="rosbag" args="--clock $(find starmap)/launch/arl_husky_floodedgrounds.bag" />
  <include file="$(find darknet_ros)/launch/darknet_ros.launch" >
    <arg name="image" value="/husky/camera/image_raw" />
    <arg name="bounding_boxes" value="/darknet_ros/bounding_boxes_0" />
  </include>
  <node pkg="nodelet" type="nodelet" name="orcvio_nodelet_manager"  args="manager" output="screen"/>
  <node pkg="nodelet" type="nodelet" name="sort_ros"
        args="load sort_ros/SortRos orcvio_nodelet_manager" >
    <param name="image_topic" value="/husky/camera/image_raw"/>
    <param name="bbox_topic" value="/darknet_ros/bounding_boxes_0" />
    <param name="tracked_bbox_topic" value="tracked_bounding_boxes_0" />
  </node>
  <node pkg="nodelet" type="nodelet" name="starmap"
        args="load starmap/Starmap orcvio_nodelet_manager"
        output="screen">
    <param name="image_topic" value="/husky/camera/image_raw" />
    <param name="bbox_topic" value="/sort_ros/tracked_bounding_boxes_0" />
    <param name="keypoint_topic" value="/starmap/keypoints_0" />
    <param name="starmap_model_path" value="$(find starmap)/models/model_cpu-jit.pth" />
    <param name="gpu_id" value="0" type="int" />
  </node>
  <!--node pkg="rqt_image_view" type="rqt_image_view" name="rqt_image_view_det"
        args=" /sort_ros/detection_image"/-->
  <!--node pkg="rqt_image_view" type="rqt_image_view" name="rqt_image_view_vis"
       args=" /starmap/visualization"/--> 
  <!-- node pkg="image_view" type="extract_images" name="extract_images_sort_ros" >
    <remap from="image" to="/sort_ros/detection_image" />
    <param name="filename_format" value="/tmp/extract_images_sort_ros_%d.jpg" />
    <param name="sec_per_frame" value="0.1"  />
  </node-->
  <node pkg="image_view" type="extract_images" name="extract_images_starmap" >
    <remap from="image" to="/starmap/visualization" />
    <param name="filename_format" value="/tmp/extract_images_starmap_%04d.jpg" />
    <param name="sec_per_frame" value="0.1"  />
  </node>

</launch>
